scripts/generate_dump

```diff
@@ -1514,9 +1532,50 @@ while getopts ":xnvhzas:t:r:d" opt; do
             ;;
         /?)
             echo "Invalid option: -$OPTARG" >&2
-            exit 1
+            exit $EXT_GENERAL
             ;;
     esac
 done
 
-main
+# Check permissions before proceeding further
+if [ `whoami` != root ] && ! $NOOP;
+then
+    echo "$0: must be run as root (or in sudo)" >&2
+    exit $EXT_INVALID_ARGUMENT
+fi
+
+##
+## Attempt Locking
+##
+
+if mkdir "${LOCKDIR}" &>/dev/null; then
+    trap 'handle_exit' EXIT
+    echo "$$" > "${PIDFILE}"
+    # This handler will exit the script upon receiving these interrupts
+    # Trap configured on EXIT will be triggered by the exit from handle_signal function
+    trap 'handle_signal' SIGINT SIGHUP SIGQUIT SIGTERM
+    echo "Lock succesfully accquired and installed signal handlers"
+    # Proceed with the actual code
+    main
+else
+    # lock failed, check if the other PID is alive
+    PID_PROG="$(cat "${PIDFILE}")"
+
+    if [ $? != 0 ]; then
+        # Another instance is probably about to remove the lock or PIDfile doesn't exist
+        # Exit and let user decide if they want to restart 
+        sleep "$(( $RANDOM % 2 + 1)).$(( $RANDOM % 999 ))"
+        $RM $V -rf ${LOCKDIR}
+        exit $EXT_RETRY
+    fi
+
+    if ! kill -0 $PID_PROG &>/dev/null; then
+        # Lock is stale
+        echo "Removing stale lock of nonexistant PID ${PID_PROG}"
+        rm_lock_and_restart
```

Comment:
>rm_lock_and_restart [](http://example.com/codeflow?start=8&length=19)

Also return EXT_RETRY?

Reply:
Will handle, i thought you only meant for the non-existent PID file case

Reply:
handled